### PR TITLE
Feature/login tests impl

### PR DESCRIPTION
**Pull request by tomklecha**
created implementation for loginNameValidation

### CODE REVIEW

**Comment by tomklecha** [comment]
sorry, i done some mistakes in adding the forEach function and some of the chars are still able to go through. I will repair and open again

**Comment by tomklecha** [comment]
added changed method for loginNameValidation, please @Canato have a look.
I have some idea for password validation, but i need to play a bit more how to write it more smooth.
Sorry it taking me so long. But Easter is coming to catch up !

File: features/login/src/main/java/com/can_apps/login/core/LoginInteractor.kt [hunk 1]
```diff
@@ -2,15 +2,26 @@ package com.can_apps.login.core
...
     override fun passwordValidation(password: LoginPasswordDomain): Boolean {
         TODO()
     }
 
     override fun loginNameValidation(loginName: LoginNameDomain): Boolean {
-        TODO()
+
+        val loginSize = loginName.value.length
+        if (loginSize < 3)
+            return false
```
Comment (Canato) [inline features/login/src/main/java/com/can_apps/login/core/LoginInteractor.kt:16]: if you don't put `{ }` suggestion to put in one line `if (loginSize < 3) return false`

File: features/login/src/main/java/com/can_apps/login/core/LoginInteractor.kt [hunk 2]
```diff
@@ -2,15 +2,26 @@ package com.can_apps.login.core
...
+
+        val pattern = Regex(pattern = "[a-zA-Z]{$loginSize}")
+        if (pattern.matches(loginName.value)) {
```
Comment (Canato) [inline features/login/src/main/java/com/can_apps/login/core/LoginInteractor.kt:19]: `if (loginName.value.matches("[a-zA-Z]+")) return true ` check if this pass in the tests, if yes we can get rid off `loginSize`variable and do `if (loginName.value.length) return false` on line 14.

File: features/login/src/test/java/com/can_apps/login/core/LoginInteractorTest.kt [hunk 3]
```diff
@@ -50,6 +50,32 @@ internal class LoginInteractorTest {
...
+        val result = interactor.loginNameValidation(loginNameDomain)
+
+        //THEN
+        assertFalse(result)
+    }
+
+    @Test
+    fun `GIVEN login with empty characters , WHEN validate, THEN return false`() {
+        //GIVEN
+        val login = "Tom @sz"
```
Comment (Canato) [inline features/login/src/test/java/com/can_apps/login/core/LoginInteractorTest.kt:69]: nice! let's have a case with space.
`val login = "Tom  asz"`

Comment (tomklecha) [reply features/login/src/test/java/com/can_apps/login/core/LoginInteractorTest.kt]: done !

File: features/login/src/main/java/com/can_apps/login/core/LoginInteractor.kt (hunk 2 above)
Comment (tomklecha) [reply features/login/src/main/java/com/can_apps/login/core/LoginInteractor.kt]: done - thank you for the suggestion - i was looking for something like that

File: features/login/src/main/java/com/can_apps/login/core/LoginInteractor.kt (hunk 1 above)
Comment (tomklecha) [reply features/login/src/main/java/com/can_apps/login/core/LoginInteractor.kt]: that was old file, now updated

File: features/login/src/main/java/com/can_apps/login/core/LoginInteractor.kt [hunk 4]
```diff
@@ -2,19 +2,38 @@ package com.can_apps.login.core
...
+                patternUpperCase.containsMatchIn(password.value) &&
+                patternDigit.containsMatchIn(password.value) &&
+                pattern.matches(password.value))
+
     }
 
     override fun loginNameValidation(loginName: LoginNameDomain): Boolean {
-        TODO()
+
+        val pattern = Regex(pattern = "[a-zA-Z]+")
```
Comment (Canato) [inline features/login/src/main/java/com/can_apps/login/core/LoginInteractor.kt:26]: put this after the If condition

File: features/login/src/main/java/com/can_apps/login/core/LoginInteractor.kt [hunk 5]
```diff
@@ -2,19 +2,38 @@ package com.can_apps.login.core
...
+) : LoginContract.Interactor {
 
+    private val patternLowerCase = Regex(pattern = "[a-z]")
+    private val patternUpperCase = Regex(pattern = "[A-Z]")
+    private val patternDigit = Regex(pattern = "[0-9]")
 
     override fun passwordValidation(password: LoginPasswordDomain): Boolean {
-        TODO()
+
+        val pattern = Regex(pattern = "[a-zA-Z0-9]+")
```
Comment (Canato) [inline features/login/src/main/java/com/can_apps/login/core/LoginInteractor.kt:13]: put this after the If condition

**Comment by tomklecha** [comment]
@Canato i need your help. I dont know what to do with this GitGuardian. I changed in last commit two lines of code, and now it dropped this.

**Pull request merged**